fundingmanager.go

```diff
@@ -585,21 +708,226 @@ func (f *fundingManager) handleFundingComplete(fmsg *fundingCompleteMsg) {
+
+	fundingPoint := completeChan.FundingOutpoint
+	fndgLog.Infof("ChannelPoint(%v) is now active",
+		fundingPoint)
+
+	completeChan.IsPending = false
+	err := f.cfg.Wallet.ChannelDB.MarkChannelAsOpen(fundingPoint)
+	if err != nil {
+		fndgLog.Errorf("error setting channel pending flag to false: "+
+			"%v", err)
+		return
+	}
+
+	// Finally, create and officially open the payment channel!
+	// TODO(roasbeef): CreationTime once tx is 'open'
+	channel, err := lnwallet.NewLightningChannel(f.cfg.Wallet.Signer,
+		f.cfg.Notifier, completeChan)
+	if err != nil {
+		fndgLog.Errorf("error creating new lightning channel: %v", err)
+		return
+	}
+
+	// Now that the channel is open, we need to notify a number of
+	// parties of this event.
+
+	// First we send the newly opened channel to the source server
+	peer, err := f.cfg.FindPeer(completeChan.IdentityPub)
+	if err != nil {
+		fndgLog.Errorf("Unable to find peer: %v", err)
+		return
+	}
+
+	newChanDone := make(chan struct{})
+	newChanMsg := &newChannelMsg{
+		channel: channel,
+		done:    newChanDone,
+	}
+	peer.newChannels <- newChanMsg
+
+	<-newChanDone
+
+	// Close the active channel barrier signalling the
+	// readHandler that commitment related modifications to
+	// this channel can now proceed.
+	f.barrierMtx.Lock()
+	fndgLog.Tracef("Closing chan barrier for ChannelPoint(%v)", fundingPoint)
+	close(f.newChanBarriers[*fundingPoint])
+	delete(f.newChanBarriers, *fundingPoint)
+	f.barrierMtx.Unlock()
+
+	// Afterwards we send the breach arbiter the new channel so it
+	// can watch for attempts to breach the channel's contract by
+	// the remote party.
+	f.cfg.ArbiterChan <- channel
+
+	// With the block height and the transaction index known, we
+	// can construct the compact chainID which is used on the
+	// network to unique identify channels.
+	chanID := lnwire.ChannelID{
+		BlockHeight: confDetails.BlockHeight,
+		TxIndex:     confDetails.TxIndex,
+		TxPosition:  uint16(fundingPoint.Index),
+	}
+
+	// When the funding transaction has been confirmed, the FundingLocked
+	// message is sent to the remote peer so that the existence of the
+	// channel can be announced to the network.
+	fundingLockedMsg := lnwire.NewFundingLocked(fundingPoint, chanID,
+		f.fakeProof.bitcoinSig, f.fakeProof.bitcoinSig, f.cfg.IDKey)
+	f.cfg.SendToPeer(completeChan.IdentityPub, fundingLockedMsg)
+
+	close(doneChan)
+	return
+}
+
+// processFundingLocked sends a message to the fundingManager allowing it to finish
+// the funding workflow.
+func (f *fundingManager) processFundingLocked(msg *lnwire.FundingLocked,
+	peerAddress *lnwire.NetAddress) {
+	f.fundingMsgs <- &fundingLockedMsg{msg, peerAddress}
+}
+
+// handleFundingLocked finalizes the channel funding process and enables the channel
+// to enter normal operating mode.
+func (f *fundingManager) handleFundingLocked(fmsg *fundingLockedMsg) {
+	fundingPoint := *fmsg.msg.ChannelOutpoint
+	channel, err := f.cfg.FindChannel(fundingPoint)
```

Comment:
Optionally, we can retain a pointer to the channel in memory for the reservation, and check that for existence before checking the database for the channel. In the case of a restart, the pointer would be `nil`, but in all other cases, it should be populted.